code/modules/robotics/bot/secbot.dm

```diff
@@ -1152,16 +1156,29 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
 		src.name = "helmet/signaler/prox sensor/robot arm assembly"
 		src.overlays += image('icons/obj/bots/aibots.dmi', "hs_arm")
 		qdel(W)
-
-	else if ((istype(W, /obj/item/baton)) && (src.build_step >= 3))
-		src.build_step++
-		boutput(user, "You complete the Securitron! Beep boop.")
-		var/obj/machinery/bot/secbot/S = new /obj/machinery/bot/secbot(get_turf(src))
-		S.beacon_freq = src.beacon_freq
-		S.hat = src.hat
-		S.name = src.created_name
-		qdel(W)
-		qdel(src)
+		
+	else if (istype(W, /obj/item/rods) && src.build_step == 3)
+		if (W.amount < 1)
+			boutput(user, "You need a non-zero amount of rods. How did you even do that?")
+		else
+			src.build_step++
+			boutput(user, "You add a rod to [src]'s robot arm!")
+			src.name = "helmet/signaler/prox sensor/robot arm/rod assembly"
+			src.overlays += image('icons/obj/bots/aibots.dmi', "hs_rod")
+			W.amount -= 1
+
+	else if (istype(W, /obj/item/cable_coil) && src.build_step >= 4)
+		if (W.amount <= 4)
+			boutput(user, "You need a longer length of cable! A length of five should be enough.")
+		else
+			src.build_step++
+			boutput(user, "You add the wires to the rod, completing the Securitron! Beep boop.")
+			var/obj/machinery/bot/secbot/S = new /obj/machinery/bot/secbot(get_turf(src))
+			S.beacon_freq = src.beacon_freq
+			S.hat = src.hat
+			S.name = src.created_name
+			W.amount -= 5
```

Comment:
```suggestion
    var/obj/item/cable_coil/C = W
		if (!C.use(5))
			boutput(user, "You need a longer length of cable! A length of five should be enough.")
		else
			src.build_step++
			boutput(user, "You add the wires to the rod, completing the Securitron! Beep boop.")
			var/obj/machinery/bot/secbot/S = new /obj/machinery/bot/secbot(get_turf(src))
			S.beacon_freq = src.beacon_freq
			S.hat = src.hat
			S.name = src.created_name
```

use the use() proc; improper use of use() is how we get cable coils not deleting themselves despite not having any cable in them

Reply:
Didn't know that, thanks for letting me know that function exists!